plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerContainerNameGenerator.java

```diff
@@ -0,0 +1,128 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.machine;
+
+import org.eclipse.che.api.core.model.machine.Machine;
+import org.eclipse.che.api.core.model.machine.MachineConfig;
+import org.eclipse.che.commons.env.EnvironmentContext;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class is used for generation docker container name or parsing important information from docker container name.
+ *
+ * @author Alexander Andrienko
+ */
+public class DockerContainerNameGenerator {
+    private static final String WORKSPACE_ID_GROUP = "(?<workspaceId>workspace[0-9a-z]+)";
+    private static final String MACHINE_ID_GROUP   = "(?<machineId>machine[0-9a-z]+)";
+
+    private static final String  CONTAINER_NAME_REGEX   = "^" + WORKSPACE_ID_GROUP + "_" + MACHINE_ID_GROUP + "_.+$";
+    private static final Pattern CONTAINER_NAME_PATTERN = Pattern.compile(CONTAINER_NAME_REGEX);
+
+    /**
+     * Return generated name for docker container. Method generate name for docker container in format:
+     * <br><p>workspaceId + "_" + machineId + "_" + userName +"_" + machineName</p>
+     * <b>Notice: if userName or machineName contains incorrect symbols for creation docker container, then we skip this symbols</b>
+     *
+     * @param workspaceId
+     *         unique workspace id, see more (@link WorkspaceConfig#getId)
+     * @param machineId
+     *         unique machine id, see more {@link Machine#getId()}
+     * @param machineName
+     *         name of the workspace machine, see more {@link MachineConfig#getName()}
+     */
+    public String generateContainerName(String workspaceId, String machineId, String machineName) {
+        String userName = EnvironmentContext.getCurrent().getUser().getName();
```

Comment:
Don't you think that it is better to pass username as an argument to avoid reliance on EnvironmentContext in that component?